lib/service.go

```diff
@@ -72,52 +65,86 @@ func makeRequest(url string, method string, params []interface{}) ([]byte, error
 		return nil, err
 	}
 
-	return ioutil.ReadAll(resp.Body)
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRPCResponse(respBody)
+}
+
+type rpcxResponse struct {
+	url string
+	err error
+	res *rpcResponse
+}
+
+// makeRequests performs the requst in parallel to all URLS
+func makeRequests(method string, args []interface{}, urls []string) (ret map[string]*rpcxResponse) {
+	ret = make(map[string]*rpcxResponse)
+	results := make(chan *rpcxResponse, len(urls))
+
+	// Start a goroutine for each URL to make requests concurrently
+	for _, url := range urls {
+		go func(url string) {
+			res := &rpcxResponse{url: url}
+			res.res, res.err = makeRequest(url, method, args)
+			results <- res
+		}(url)
+	}
+
+	// Collect responses from the goroutines
+	for i := 0; i < cap(results); i++ {
+		res := <-results
+		ret[res.url] = res
+	}
+
+	return ret
```

Comment:
I think you don't need to return `ret`..bcz ret is already a  named return value. I have seen this in many of our repos(Just noticed but not a problem)
```suggestion
	return
```